### DISCUSSION TITLE
Show fingermap to help with learning

### DISCUSSION THREAD

**Discussion by torjacob** [monkeytypegame/monkeytype#4243] [Ideas]
Showing a "fingermap" / which finger to use when pressing next key could be useful for beginners.

Given that Monkeytype already has a system for defining a fingermap, a version where a user could either select from a list of "standard" hand positions, or define one for themselves through the fingermap tool and then select it would probably be a good solution.

In essence, something along the lines of how [typingstudy][ts] shows which finger to use:

![tsui](https://gcdnb.pbrd.co/images/0SzrMOOar6yM.png?o=1)

If this is something that has been brought up before I apologize, however i couldn't find anything similar through searching the repo.

[ts]: https://www.typingstudy.com/

**Comment by tajhans** [comment]
Keymap is a feature that shows the keyboard on screen. There are 3 modes you can pick from: static, next, and react. Static just displays the keyboard, and react shows whatever key you just pressed, but next will show the next key you have to press. Next functions just like that feature.
![IMG_0187](https://github.com/monkeytypegame/monkeytype/assets/83455454/c8ba8355-97b0-4abc-9165-b04b6a1db11b)
![IMG_0189](https://github.com/monkeytypegame/monkeytype/assets/83455454/2efc6600-4ea3-4bbf-9a40-668c7ce3f3d7)

**Reply by torjacob** [reply]
I'm aware of the keymap / the modes it offers, and think it's a great feature. I realize now that i could've made this more clear in the feature idea.

Although _Next_ shows which key to press, it supplies no visual as to which finger to use on the indicated key. Any user who is new to typing might have either "incorrect" or no technique/finger placement. Only showing the upcoming key therefore doesn't supply the user with enough information to improve effectively. It could also serve to engrain bad habits.

What I'm suggesting is the option to add an indication of which finger to use, as illustrated in the [screengrab in my original post](https://camo.githubusercontent.com/1b49511f4d27dafd85b9641123e57f0f7b44f3582adc485e24aa6b33c13c48e8/68747470733a2f2f6763646e622e706272642e636f2f696d616765732f30537a724d4f4f617236794d2e706e673f6f3d31) (hands with glowing finger). This could either be separate, e.g. choice to only indicate fingers and either show or hide keymap - or as an additional option to the keymap (although I reckon the first option would be better).
